lib/Controller/ConfigController.php

```diff
@@ -13,31 +13,35 @@
 
 use DateTime;
 use OCA\Gitlab\AppInfo\Application;
+use OCA\Gitlab\Model\AdminConfig;
+use OCA\Gitlab\Model\UserConfig;
 use OCA\Gitlab\Reference\GitlabReferenceProvider;
+use OCA\Gitlab\Service\ConfigService;
 use OCA\Gitlab\Service\GitlabAPIService;
 use OCP\AppFramework\Controller;
 use OCP\AppFramework\Http;
 use OCP\AppFramework\Http\DataResponse;
 use OCP\AppFramework\Http\RedirectResponse;
 use OCP\AppFramework\Http\TemplateResponse;
 use OCP\AppFramework\Services\IInitialState;
-use OCP\IConfig;
 use OCP\IL10N;
 use OCP\IRequest;
 use OCP\IURLGenerator;
 use OCP\PreConditionNotMetException;
 
 class ConfigController extends Controller {
 
-	public function __construct(string $appName,
+	public function __construct(
+		string $appName,
 		IRequest $request,
-		private IConfig $config,
+		private ConfigService $config,
 		private IURLGenerator $urlGenerator,
 		private IL10N $l,
 		private IInitialState $initialStateService,
 		private GitlabAPIService $gitlabAPIService,
 		private GitlabReferenceProvider $gitlabReferenceProvider,
-		private ?string $userId) {
+		private string $userId,
```

Comment:
are we sure this will never be called without a userId?

Reply:
Yes (also none of the routes are public pages).

Reply:
cool